File: src/main/java/dev/fringe/crud/model/Category.java
package dev.fringe.crud.model;

import lombok.Data;
import org.apache.commons.lang3.builder.ToStringBuilder;
import org.apache.commons.lang3.builder.ToStringStyle;

import javax.persistence.*;
import java.util.Set;

@Entity
@Table(name = "CATEGORY")
public class Category {

    @Id
    @Column(name = "CATEGORY_ID")
    @GeneratedValue
    private long id;
    private String name;

    @OneToMany(mappedBy = "category", cascade = CascadeType.ALL)
    private Set<Product> products;

    public Category() {
    }

    public Category(String name) {
        this.name = name;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Set<Product> getProducts() {
        return products;
    }

    public void setProducts(Set<Product> products) {
        this.products = products;
    }
    public String toString() {
        return ToStringBuilder
                .reflectionToString(this, ToStringStyle.MULTI_LINE_STYLE);
    }
}

File: src/main/scala/dev/fringe/crud/model/Category.scala
package dev.fringe.crud.model

import javax.persistence._

@Entity
@Table(name = "CATEGORY")
class Category (cname : String){

    @Id
    @Column(name = "CATEGORY_ID")
    @GeneratedValue
    var id : Long = _
    var name : String = cname

    @OneToMany(mappedBy = "category", cascade = Array(CascadeType.ALL))
    var products : java.util.Set[Product] = _

    override def toString = s"Category($id, $name, $products)"
}